src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -22127,6 +22127,1242 @@ void Main(string? s)
+                // (27,15): warning CS8604: Possible null reference argument for parameter 't' in 'void Program.F3<T>(T? t)'.
+                //         F3<T>(t3); // 7
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "t3").WithArguments("t", "void Program.F3<T>(T? t)").WithLocation(27, 15));
+        }
+
+        [Fact]
+        public void DisallowNull_03()
+        {
+            var source =
+@"using System.Runtime.CompilerServices;
+class Program
+{
+    static void F1([DisallowNull]string s) { }
+    static void F2([DisallowNull]string? s) { }
+    static void M1(string s1)
+    {
+        F1(s1);
+        F2(s1);
+        s1 = null; // 1
+        F1(s1); // 2
+        F2(s1); // 3
+    }
+    static void M2(string? s2)
+    {
+        F1(s2); // 4
+        F2(s2); // 5
+        if (s2 == null) return;
+        F1(s2);
+        F2(s2);
+    }
+}";
+            var comp = CreateCompilation(new[] { DisallowNullAttributeDefinition, source }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (10,14): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         s1 = null; // 1
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(10, 14),
+                // (11,12): warning CS8604: Possible null reference argument for parameter 's' in 'void Program.F1(string s)'.
+                //         F1(s1); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "s1").WithArguments("s", "void Program.F1(string s)").WithLocation(11, 12),
+                // (12,12): warning CS8604: Possible null reference argument for parameter 's' in 'void Program.F2(string? s)'.
+                //         F2(s1); // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "s1").WithArguments("s", "void Program.F2(string? s)").WithLocation(12, 12),
+                // (16,12): warning CS8604: Possible null reference argument for parameter 's' in 'void Program.F1(string s)'.
+                //         F1(s2); // 4
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "s2").WithArguments("s", "void Program.F1(string s)").WithLocation(16, 12),
+                // (17,12): warning CS8604: Possible null reference argument for parameter 's' in 'void Program.F2(string? s)'.
+                //         F2(s2); // 5
+                Diagnostic(ErrorCode.WRN_NullReferenceArgument, "s2").WithArguments("s", "void Program.F2(string? s)").WithLocation(17, 12));
+        }
+
+        [Fact]
+        public void DisallowNull_04()
+        {
+            var source =
+@"using System.Runtime.CompilerServices;
+class Program
+{
+    static void F1([DisallowNull]int i) { }
+    static void F2([DisallowNull]int? i) { }
+    static void M1(int i1)
+    {
+        F1(i1);
+        F2(i1);
+    }
+    static void M2(int? i2)
+    {
+        F2(i2);
```

Comment:
>F2(i2); [](start = 8, length = 7)

I expected a warning here (we're passing a maybe-null state into a parameter marked with `[DisallowNull]`)